util/update-pcluster-configs.py

```diff
@@ -0,0 +1,490 @@
+#!/usr/bin/python
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file.
+# This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, express or implied.
+# See the License for the specific language governing permissions and limitations under the License.
+import collections
+import json
+import logging
+import re
+from abc import ABC, abstractmethod
+from urllib.request import urlopen
+
+import argparse
+import boto3
+from botocore.exceptions import ClientError, EndpointConnectionError
+
+from jsonschema import validate
+
+logging.basicConfig(level=logging.INFO, format="%(asctime)s %(levelname)s [%(name)s] %(message)s")
+
+PARTITIONS = ["commercial", "china", "govcloud"]
+CONFIG_FILES = ["instances", "feature_whitelist"]
+PARTITION_TO_MAIN_REGION = {"commercial": "us-east-1", "govcloud": "us-gov-west-1", "china": "cn-north-1"}
+FILE_TO_S3_PATH = {"instances": "instances/instances.json", "feature_whitelist": "features/feature_whitelist.json"}
+
+
+class S3JsonDocumentManager:
+    def __init__(self, region, credentials=None):
+        self._region = region
+        self._credentials = credentials or {}
+
+    def download(self, s3_bucket, document_s3_path):
+        try:
+            s3 = boto3.resource("s3", region_name=self._region, **self._credentials)
+            instances_file_content = s3.Object(s3_bucket, document_s3_path).get()["Body"].read()
+            return json.loads(instances_file_content)
+        except Exception as e:
+            logging.error(
+                "Failed when downloading file %s from bucket %s in region %s with error %s",
+                document_s3_path,
+                s3_bucket,
+                self._region,
+                e,
```

Comment:
maybe the error message is enough? Also in the next Exception

Reply:
not sure what the exception message is going to be so I'd rather leave and extra clear logging message